### PR TITLE
MAT-156 span sync using system call

### PR DESCRIPTION
**Pull request by jdkanani**

### CODE REVIEW

**Comment by 0xAshish** [comment]
just for general discussion, I want all the blocks to be signed by Block Producer as discussed with @vaibhavchellani. Ofc this system tx means 2/3+1 consensus but for plasma security and blame reasons.

**Comment by jdkanani** [comment]
@0xAshish Yes, each block already contains block producer's sign. Not sure what do you mean in case of system TX. Can you elaborate more?

**Comment by 0xAshish** [comment]
> each block already contains block producer's sign

then it's cool, we need this in the context of system tx it clicked to me that if we aren't doing then we need it for systemTx just for slashing and fraud proofs/accounting.

File: contracts/StateReceiver.sol [hunk 1]
```diff
@@ -3,58 +3,65 @@ pragma solidity ^0.5.11;
...
-    bytes calldata txBytes,
-    bytes calldata proof
+  // add pending state
+  function proposeState(
+    uint256 stateId
   ) external {
-    // validator validator set
-    validateValidatorSet(vote, sigs, txBytes, proof);
+    // check if sender is validator
+    isValidator(msg.sender);
```
Comment (atvanguard) [inline contracts/StateReceiver.sol]: This needs to be placed in a `require`.

Comment (0xAshish) [reply contracts/StateReceiver.sol]: good catch.

Comment (0xAshish) [reply contracts/StateReceiver.sol]: resolved

**Pull request merged**